posthog/api/utils.py

```diff
@@ -361,84 +361,31 @@ def select_event_definitions(fields: str):
         """
 
     # Only return event definitions
-    if (
-        event_type == CombinedEventType.EVENT
-        or event_type == CombinedEventType.EVENT_CUSTOM
-        or event_type == CombinedEventType.EVENT_POSTHOG
-    ):
-        raw_event_definition_fields = ",".join(event_definition_fields)
-        ordering = (
-            "ORDER BY last_seen_at DESC NULLS LAST, query_usage_30_day DESC NULLS LAST, name ASC"
-            if is_enterprise
-            else "ORDER BY name ASC"
-        )
-
-        if event_type == CombinedEventType.EVENT_CUSTOM:
-            shared_conditions += " AND posthog_eventdefinition.name NOT LIKE %(is_posthog_event)s"
-        if event_type == CombinedEventType.EVENT_POSTHOG:
-            shared_conditions += " AND posthog_eventdefinition.name LIKE %(is_posthog_event)s"
-
-        return (
-            f"""
-                {select_ee_event_definitions(raw_event_definition_fields)}
-                {shared_conditions}
-                {ordering}
-            """
-            if is_enterprise
-            else f"""
-                {select_event_definitions(raw_event_definition_fields)}
-                {shared_conditions}
-                {ordering}
-            """
-        )
```

Comment:
I only kept this `if` condition and got rid of the rest, which used magical SQL ✨  to get actions and event definitions in one combined query.